### PR TITLE
dev-util/pkgconf: bump to version 1.0.1

### PR DESCRIPTION
**Pull request by doughdemon**
@Elizafox

### CODE REVIEW

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="git://github.com/pkgconf/pkgconf.git"
+	inherit autotools git-r3 multilib-minimal
+else
+	inherit multilib-minimal
```
Comment (SoapGentoo) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: Take the `inherit multilib-minimal` out of the loop and after it

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	)"
+
+MULTILIB_CHOST_TOOLS=(
+	/usr/bin/pkgconf
+)
+
+src_prepare() {
+	[[ -e configure ]] || eautoreconf
+
+	default
```
Comment (SoapGentoo) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: I would swap these - currently, there's no way for users to patch the `configure.ac` if they wish to.

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="git://github.com/pkgconf/pkgconf.git"
```
Comment (mgorny) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: Please add a https:// source as well, and as the first (preferred) one. This gives people the little bit of extra security via HTTPS certificate verification.

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	inherit autotools git-r3 multilib-minimal
+else
+	inherit multilib-minimal
+	SRC_URI="https://github.com/pkgconf/pkgconf/releases/download/${P}/${P}.tar.gz"
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~x64-freebsd ~x86-freebsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc64-solaris ~x64-solaris"
```
Comment (mgorny) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: Wrap this, please.

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		!dev-util/pkg-config-lite
+		!dev-util/pkgconfig-openbsd[pkg-config]
+	)"
+
+MULTILIB_CHOST_TOOLS=(
+	/usr/bin/pkgconf
+)
+
+src_prepare() {
+	[[ -e configure ]] || eautoreconf
```
Comment (mgorny) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: You are mixing two different conditions for relevant things. In other words:
1. autotools is inherited if ${PV}==9999,
2. but eautoreconf is called if configure does not exist.

Use the same condition throughout the whole file.

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	ECONF_SOURCE=${S} \
+	econf $(use_enable strict)
+}
+
+multilib_src_install() {
+	default
+	use pkg-config \
+		&& dosym pkgconf /usr/bin/pkg-config \
+		|| rm "${ED}"/usr/share/aclocal/pkg.m4 \
+		|| die
```
Comment (mgorny) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: Don't do crazy multi-branch conditionals like this. Use if;else.

**Comment by doughdemon** [comment]
The commit is updated to include all requested changes. Thanks for the reviews!

File: dev-util/pkgconf/pkgconf-1.0.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	econf $(use_enable strict)
+}
+
+multilib_src_install() {
+	default
+
+	if use pkg-config; then
+		dosym pkgconf /usr/bin/pkg-config
+	else
+		rm "${ED}"/usr/share/aclocal/pkg.m4 || die
```
Comment (mgorny) [inline dev-util/pkgconf/pkgconf-1.0.1.ebuild]: Nitpick: ED already ends with a slash, so you have a double slash here. Either remove the slash or use `${ED%/}`

**Comment by doughdemon** [comment]
removed the slash after ${ED}.

**Comment by mgorny** [comment]
@gentoo/proxy-maint

File: dev-util/pkgconf/pkgconf-9999.ebuild [hunk 8]
```diff
@@ -49,8 +53,10 @@ multilib_src_configure() {
...
 	default
-	use pkg-config \
-		&& dosym pkgconf /usr/bin/pkg-config \
-		|| rm "${ED}"/usr/share/aclocal/pkg.m4 \
-		|| die
+
+	if use pkg-config; then
+		dosym pkgconf /usr/bin/pkg-config
+	else
+		rm "${ED}"/usr/share/aclocal/pkg.m4 || die
```
Comment (SoapGentoo) [inline dev-util/pkgconf/pkgconf-9999.ebuild]: You still have the double-slash here. My style is to prefer `"${ED%/}"/bla/foo`, as that will always be guaranteed to work and be single-slashed, even if Gentoo decides to get rid of slash-terminated `ED` some day.

**Comment by doughdemon** [comment]
ok, then let me keep it as it is. As it is now, the ebuild is the most
readable. The double-slash won't be visible in the build.log anyway.

**Comment by doughdemon** [comment]
> Furthermore, it's not about what's visible but about using `//` that can break some systems.

POSIX specifies that multiple slashes have the same effect as a single
slash. The only POSIX-like system I could find not conforming to this
is cygwin, which treats leading double slashes specially.

I guess that on prefix systems $ED will never be '/' so this does not
apply to the path we are discussing, but anyway I've changed the path.

**Comment by Polynomial-C** [comment]
What's the status here? Why is this PR rotting for so long?

**Comment by doughdemon** [comment]
The proxied maintainer @Elizafox hasn't reacted, yet.

Let me also say that I somewhat recently added a git:// url to
`EGI_REPO_URI`. With that, I think that all concerns are addressed.

**Pull request closed without merging**